### PR TITLE
Grant fetch timestamp is now reset when no grants are available

### PR DESCRIPTION
**Pull request by jasonrsadler**
Core Implementation: https://github.com/brave/brave-core/pull/807

Also does not attempt retries on error since 404s are considered errors.

### CODE REVIEW

File: src/ledger_impl.cc [hunk 1]
```diff
@@ -573,7 +573,7 @@ void LedgerImpl::OnGrant(ledger::Result result, const braveledger_bat_helper::GR
 
   grant.promotionId = properties.promotionId;
   last_grant_check_timer_id_ = 0;
-  RefreshGrant(result != ledger::Result::LEDGER_OK);
+  RefreshGrant(false);
```
Comment (NejcZdovc) [inline src/ledger_impl.cc]: Can you give me some more background why this was changed? Why we don't want to retry on error?

Comment (jasonrsadler) [reply src/ledger_impl.cc]: If no grants are available it returns 404 error. We don't want to use retry interval if no grants are available, we would just check on the next 24 hour interval.

Comment (NejcZdovc) [reply src/ledger_impl.cc]: but what happens if there is server error and not 404?

Comment (jasonrsadler) [reply src/ledger_impl.cc]: We try again in 24 hours.

Comment (NejcZdovc) [reply src/ledger_impl.cc]: let's do server errors and our errors via retries and 404 via 24 cycle

Comment (jasonrsadler) [reply src/ledger_impl.cc]: No problem 😁

Comment (jasonrsadler) [reply src/ledger_impl.cc]: Done

**Pull request merged**